Framework/Kernel/src/PropertyNexus.cpp

```diff
@@ -0,0 +1,219 @@
+             const std::vector<Kernel::DateAndTime> &times) {
+  std::vector<NumT> values;
+  file->getData(values);
+  if (times.empty()) {
+    if (values.size() == 1) {
+      return Mantid::Kernel::make_unique<PropertyWithValue<NumT>>(name,
+                                                                  values[0]);
+    } else {
+      return Mantid::Kernel::make_unique<ArrayProperty<NumT>>(name, values);
+    }
+  } else {
+    auto prop = Mantid::Kernel::make_unique<TimeSeriesProperty<NumT>>(name);
+    prop->addValues(times, values);
+    return std::unique_ptr<Property>(std::move(prop));
+  }
+}
+
+/** Helper method to create a time series property from a boolean
+*
+* @param file :: nexus file handle
+* @param name :: name of the property being created
+* @param times :: vector of times, empty = single property with value
+* @return Property *
+*/
+std::unique_ptr<Property>
+makeTimeSeriesBoolProperty(::NeXus::File *file, const std::string &name,
+                           const std::vector<Kernel::DateAndTime> &times) {
+  std::vector<uint8_t> savedValues;
+  file->getData(savedValues);
+  const size_t nvals = savedValues.size();
+  std::vector<bool> realValues(nvals);
+  for (size_t i = 0; i < nvals; ++i) {
+    realValues[i] = (savedValues[i] != 0);
+  }
+  auto prop = Mantid::Kernel::make_unique<TimeSeriesProperty<bool>>(name);
+  prop->addValues(times, realValues);
+  return std::unique_ptr<Property>(std::move(prop));
+}
+
+/** Make a string/vector\<string\> property */
+std::unique_ptr<Property>
+makeStringProperty(::NeXus::File *file, const std::string &name,
+                   const std::vector<Kernel::DateAndTime> &times) {
+  std::vector<std::string> values;
+  if (times.empty()) {
+    std::string bigString = file->getStrData();
+    return Mantid::Kernel::make_unique<PropertyWithValue<std::string>>(
+        name, bigString);
+  } else {
+    if (file->getInfo().dims.size() != 2)
+      throw std::runtime_error("NXlog loading failed on field " + name +
+                               ". Expected rank 2.");
+    int64_t numStrings = file->getInfo().dims[0];
+    int64_t span = file->getInfo().dims[1];
+    auto data = Mantid::Kernel::make_unique<char[]>(numStrings * span);
+    file->getData(data.get());
+    values.reserve(static_cast<size_t>(numStrings));
+    for (int64_t i = 0; i < numStrings; i++)
+      values.emplace_back(data.get() + i * span);
+
+    auto prop =
+        Mantid::Kernel::make_unique<TimeSeriesProperty<std::string>>(name);
+    prop->addValues(times, values);
+    return std::unique_ptr<Property>(std::move(prop));
+  }
+}
+
+//----------------------------------------------------------------------------------------------
+/** Opens a NXlog group in a nexus file and
+ * creates the correct Property object from it.
+ *
+ * @param file :: NXS file handle
+ * @param group :: name of NXlog group to open
+ * @return Property pointer
+ */
+std::unique_ptr<Property> loadProperty(::NeXus::File *file,
```

Comment:
Not directly related to this PR, but I think that [LoadNexusLogs::createTimeSeries](https://github.com/mantidproject/mantid/blob/master/Framework/DataHandling/src/LoadNexusLogs.cpp) has much overlap with this and is an (almost?) duplicate.